jetty-webapp/src/test/java/org/eclipse/jetty/webapp/HttpSessionListenerTest.java

```diff
@@ -0,0 +1,61 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.webapp;
+
+import java.util.Collection;
+import javax.servlet.http.HttpSessionEvent;
+import javax.servlet.http.HttpSessionListener;
+
+import org.eclipse.jetty.server.Server;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class HttpSessionListenerTest
+{
+    public static class MyHttpSessionListener implements HttpSessionListener
+    {
+        @Override
+        public void sessionCreated(HttpSessionEvent se)
+        {
+        }
+
+        @Override
+        public void sessionDestroyed(HttpSessionEvent se)
+        {
+        }
+    }
+
+    @Test
+    public void testSessionListeners()
+    {
+        Server server = new Server();
+
+        WebAppContext wac = new WebAppContext();
+
+        wac.setServer(server);
+        server.setHandler(wac);
+        wac.addEventListener(new MyHttpSessionListener());
+
+        Collection<MyHttpSessionListener> listeners = wac.getSessionHandler().getBeans(MyHttpSessionListener.class);
+        assertNotNull(listeners);
+        assertEquals(1, listeners.size());
```

Comment:
We are just testing an aspect of the implementation here, ie that we set listeners as bean.
Do we need to test more?  Or a session listeners already tested for functionality elsewhere?? in which case is this test really needed?

Reply:
Hmmm we already have a SessionListener test, so I think this either duplicates an existing test or fills a hole in testing in the wrong spot

Reply:
Could be, i'll evaluate the other HttpSessionListener tests, see if there's either a better place or a duplicated testcase.

Reply:
This one also came from WebAppContextTest, but it wasn't testing WebAppContext, it was testing HttpSessionListener.

https://github.com/eclipse/jetty.project/blob/jetty-9.4.31.v20200723/jetty-webapp/src/test/java/org/eclipse/jetty/webapp/WebAppContextTest.java#L63-L96

Reply:
These have been moved to SessionListenerTest